src/main/java/com/networknt/schema/BaseJsonValidator.java

```diff
@@ -149,18 +149,22 @@ protected String getNodeFieldType() {
         }
         return null;
     }
-    
-	/**
-	 * This is default implementation of walk method. Its job is to call the
-	 * validate method if shouldValidateSchema is enabled.
-	 */
-	@Override
-	public Set<ValidationMessage> walk(JsonNode node, JsonNode rootNode, String at, boolean shouldValidateSchema) {
-		Set<ValidationMessage> validationMessages = new LinkedHashSet<ValidationMessage>();
-		if (shouldValidateSchema) {
-			validationMessages = validate(node, rootNode, at);
-		}
-		return validationMessages;
-	}
+
+    /**
+     * This is default implementation of walk method. Its job is to call the
+     * validate method if shouldValidateSchema is enabled.
+     */
+    @Override
+    public Set<ValidationMessage> walk(JsonNode node, JsonNode rootNode, String at, boolean shouldValidateSchema) {
+        Set<ValidationMessage> validationMessages = new LinkedHashSet<ValidationMessage>();
+        if (shouldValidateSchema) {
+            validationMessages = validate(node, rootNode, at);
+        }
+        return validationMessages;
+    }
+
+    protected boolean isPartOfMultipleType(){
+        return parentSchema.schemaPath.equals(ValidatorTypeCode.ONE_OF.getValue());
```

Comment:
`ALL_OF` and `ANY_OF` are multiple types too, it should return true in these cases.
Or change method name to avoid confusion.

Reply:
In this case maybe we have to do the same for ALL_OF and ANY_OF validator

Reply:
Thank you @guillomep. I thought I would fix ALL_OF and ANY_OF later. But for now i have renamed the method as you had suggested.